fn_phish_tank/fn_phish_tank/components/fn_phish_tank_submitt_url.py

```diff
@@ -0,0 +1,93 @@
+    def _fn_phish_tank_submitt_url_function(self, event, *args, **kwargs):
+        """Function: Function to check URL against PhishTanks(https://www.phishtank.com/) Database to see URL is flagged as phishing or not."""
+        try:
+            # Get the function parameters:
+            phish_tank_check_url = kwargs.get("phish_tank_check_url")  # text
+
+            # Get function parameters from config file.
+            phish_tank_api_url = self.options.get('phish_tank_api_url')
+            phish_tank_api_key = self.options.get('phish_tank_api_key')
+            proxy = self.options.get('proxy')
+
+            log = logging.getLogger(__name__)
+            log.info("PhishTank Check URL: %s", phish_tank_check_url)
+            log.info("PhishTank API Access URL: %s", phish_tank_api_url)
+            log.info("PhishTank API Access KEY: %s", phish_tank_api_key)
+            log.info("Proxy Server Address: %s", proxy)
+
+            yield StatusMessage("Checking Phishing Status Against URL: {}".format(phish_tank_check_url))
+
+            # Initialing the resilient result object
+            _result_obj = ResultPayload('fn_phish_tank',
+                                        function_inputs=[phish_tank_api_url, phish_tank_check_url, phish_tank_api_key])
+
+            # PhihshTank Proxy Data
+            pt_proxy = phish_tank_helper.format_proxy_data(proxy_data=proxy)
+
+            # PhishTank POST Call Data
+            pt_post_call_data = phish_tank_helper.create_post_data(phish_tank_check_url, api_key=phish_tank_api_key)
+
+            # Initialising the Request Session
+            pt_helper_class_instance = phish_tank_helper()
+            _request_session = pt_helper_class_instance.session()
+
+            # Making Post request to PhishTank Database for check Phishing status against given URL
+            _api_response = _request_session.post(phish_tank_api_url, data=pt_post_call_data, proxies=proxy)
+
+            _api_response.raise_for_status()
+
+            # Converting received response to json format
+            _api_response_json = _api_response.json()
+
+            # Converting string time to epoch format
+            _received_time_stamp = _api_response_json.get('meta').get('timestamp')
+            _epoch_time = phish_tank_helper.timestamp_to_ms_epoch(_received_time_stamp)
+
+            # Converting verified string time to epoch format if verified is true from api_response
+            _verified_status = _api_response_json.get('results').get('verified')
+            if _verified_status:
+                _received_verified_time = _api_response_json.get('results').get('verified_at')
+                _epoch_verified_time = phish_tank_helper.timestamp_to_ms_epoch(_received_verified_time)
+                _api_response_json['results']['verified_at'] = _epoch_verified_time
+
+            # Replacing string time with epoch converted time
+            _api_response_json['meta']['timestamp'] = _epoch_time
```

Comment:
I don't like the idea of modifying the original parameter. If anything, add a new one like 'timestamp_modified'

Reply:
ok.